source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -463,27 +463,33 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
 
   parsed_alpn_protocols_ = parseAlpnProtocols(config.alpnProtocols());
 
-  // To enumerate the required builtin ciphers, curves, algorithms, and
-  // versions, uncomment '#define LOG_BUILTIN_STAT_NAMES' below, and run
-  //  bazel test //test/extensions/transport_sockets/tls/... --test_output=streamed
-  //      | grep " Builtin ssl." | sort | uniq
-  // #define LOG_BUILTIN_STAT_NAMES
-  //
-  // TODO(#8035): improve tooling to find any other built-ins needed to avoid
-  // contention.
-
-  // Ciphers
-  stat_name_set_->rememberBuiltin("AEAD-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-ECDSA-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-AES128-SHA");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-CHACHA20-POLY1305");
+  // Ciphers are configured as a string delimited by ":", with equivalence
+  // groups given as "[alt1|alt2]", and exclusions preceded by "!". For the
+  // purposes of collecting stats -- we want to track all these names. We don't
+  // need to fully parse out the structure of the cipher suites -- just extract
+  // out the names. We skip exclusions as those do not show up as entire
+  // stat-name segments, so they would not match.
+  for (absl::string_view cipher_suite :
+       absl::StrSplit(config.cipherSuites(), absl::ByAnyChar(":|[]"))) {
+    if (!cipher_suite.empty() && cipher_suite[0] != '!') { // skip exclusions.
+      stat_name_set_->rememberBuiltin(cipher_suite);
+    }
+  }
+
+  // This cipher is referenced in a test, though it's not obvious how or why.
   stat_name_set_->rememberBuiltin("TLS_AES_128_GCM_SHA256");
 
   // Curves from
   // https://github.com/google/boringssl/blob/f4d8b969200f1ee2dd872ffb85802e6a0976afe7/ssl/ssl_key_share.cc#L384
+  //
+  // TODO(jmarantz): consider whether we need this hard-coded list in addition
+  // to the ones from ecdhCurves below. Note there is no harm in remembering a
+  // curve more than once.
```

Comment:
@PiotrSikora do you think I can remove the hardcoded set here and just use ecdhCurves()?

Reply:
Ideally, we would use the configured curves from `SSL_CTX`, since there are a few naming conventions and users might use `secp256r1` or `prime256v1` instead of `P-256`, in which case we wouldn't be able to log the stats for them, but it doesn't look that there is a getter for that in BoringSSL.

With that in mind, we could either use the hardcoded list as before, possibly dropping the experimental post-quantum curves from the list (`CECPQ2*`), or we could try to map all alternative names provided in `config.ecdhCurves()` to NIST names (the alternatives are listed in [`RFC4492, appendix A`](https://tools.ietf.org/search/rfc4492#appendix-A)). Either one is fine.

Reply:
I think I'll leave it as is; a small number of preconfigured curves, combined with the ones from the config, is not a problem; we just don't want to have the space unbounded.